### DISCUSSION TITLE
Create a system of "business roles": "Accountant", "Sales", ...

### DISCUSSION THREAD

**Discussion by ehuelsmann** [ledgersmb/LedgerSMB#5808] [Design]
Business roles ("Accountant", "Sales", "Admin", ...) can be composed of application roles (lsmb_<company>__account_edit, ...). With these business roles, it'd be much easier for starting users to get started. Once they have the system up and running, they can then fine-tune the application role assignment to the business roles, providing their users with the fine-tuned rights they need.

The problem is: we currently don't have anything in place to create these business roles.

**Comment by einhverfr** [comment]
See #902 for an older version (moved to github and then closed) of this same issue.

**Comment by einhverfr** [comment]
Recommend that such roles follow the name of:

[prefix]application_[rolename]

We could then filter these out and put them in a top section.

**Comment by JigmeDatse** [comment]
Would it make sense for some "business role" settings section?  Rather than pre-defining the business roles in the same interface, you could setup the roles, and the permissions separately.  So, it would be something like creating a new "user" in regard to current setup.  But it would be a "business role" user?  Or something along those lines?  Currently for basic "sales" roles, they are sitting in multiple areas of the list.

**Comment by einhverfr** [comment]
Well, if we have a way of differentiating them, we should be able to build an interface based on what we have that allows their creation and changes to permssions on them.

However I think the Pg catalogs are sufficient for this so if we settle on a naming convention, it should become possible to:
1.  List these
2.  Add new ones, and
3.  Set permissions

reusing a lot of code we already have.......

**Comment by einhverfr** [comment]
the need to have an editing interface more or less means this should be consolidated with https://sourceforge.net/p/ledger-smb/feature-requests/249/
